lib/simplecov.rb

```diff
@@ -79,9 +79,9 @@ def result
       # first, then merge the results and return those
       if use_merging
         SimpleCov::ResultMerger.store_result(@result) if @result
-        return SimpleCov::ResultMerger.merged_result
-      else
-        return @result if defined? @result
+        SimpleCov::ResultMerger.merged_result
+      elsif defined? @result
+        @result if defined? @result
```

Comment:
I like the approach, but this changes the logic, you're missing an `else SimpleCov::ResultMerger.merged_result` case here to keep it in line with the previous version, no?